app/src/main/java/dev/ukanth/ufirewall/Api.java

```diff
@@ -1422,7 +1422,7 @@ public static List<PackageInfoData> getApps(Context ctx, GetAppList appList) {
                 Matcher m = p.matcher(user.toString());
                 if (m.find() && m.groupCount() > 0) {
                     int id = Integer.parseInt(m.group(1));
-                    if (id > 0) {
+                    if (id >= 0) {
```

Comment:
uid 0 is root . That's handled separately. We don't need to add it here.

Reply:
I got the issue that at least in one of my (rooted) dev devices (I believe the one running LineageOS 18.1 (i.e. Android 11)) the only user reported at that point is the one with id 0...

Reply:
p.s. I've dropped the other commits (there are individual pulls for each of those).

Reply:
https://github.com/ukanth/afwall/blob/59d20c083cae02608a3dcf0c754af76921bdf373/app/src/main/java/dev/ukanth/ufirewall/Api.java#L257C6-L265C7 - Special apps.

Reply:
On the affected devices the listOfUids will always be empty, and therefore the packagesForUser list will empty too:
```
            if(G.supportDual()) {
                packagesForUser  = getPackagesForUser(listOfUids);
            }
```